src/raftstore/store/fsm/store.rs

```diff
@@ -989,35 +989,76 @@ impl RaftBatchSystem {
     ) -> Result<()> {
         builder.snap_mgr.init()?;
 
+        let engines = builder.engines.clone();
+        let snap_mgr = builder.snap_mgr.clone();
+        let cfg = builder.cfg.clone();
+        let store = builder.store.clone();
+        let pd_client = builder.pd_client.clone();
+        let importer = builder.importer.clone();
+
+        let apply_poller_builder = ApplyPollerBuilder::new(
+            &builder,
+            ApplyNotifier::Router(self.router.clone()),
+            self.apply_router.clone(),
+        );
+        self.apply_system
+            .schedule_all(region_peers.iter().map(|pair| pair.1.get_peer()));
+
+        let reader = LocalReader::new(&builder, region_peers.iter().map(|pair| pair.1.get_peer()));
+
+        let tag = format!("raftstore-{}", store.get_id());
+        self.system.spawn(tag, builder);
+        let mut mailboxes = Vec::with_capacity(region_peers.len());
+        let mut address = Vec::with_capacity(region_peers.len());
+        for (tx, fsm) in region_peers {
+            address.push(fsm.region_id());
+            mailboxes.push((fsm.region_id(), BasicMailbox::new(tx, fsm)));
+        }
+        self.router.register_all(mailboxes);
+        // Make sure Msg::Start is the first message each FSM received.
+        self.router
+            .send_control(StoreMsg::Start {
+                store: store.clone(),
+            })
+            .unwrap();
+        for addr in address {
+            self.router.force_send(addr, PeerMsg::Start).unwrap();
+        }
+
+        self.apply_system
```

Comment:
can we move L992-1007 here?

Reply:
@Connor1996  I'm afraid not, the `builder` and `region_peers` variables were moved at L1010 and L1013.